internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -0,0 +1,407 @@
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const quickstartTemplate = "Now you can connect to your Atlas cluster with: mongo -u %s -p %s %s\n"
+
+const (
+	replicaSet        = "REPLICASET"
+	diskSizeGB        = 10
+	mdbVersion        = "4.2"
+	shards            = 1
+	tier              = "M10"
+	members           = 3
+	zoneName          = "Zone 1"
+	accessListComment = "IP added with mongocli atlas quickstart"
+	atlasAdmin        = "atlasAdmin"
+	none              = "NONE"
+	passwordLength    = 12
+	maxRandNum        = 10000
+)
+
+type Opts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	clusterName      string
+	provider         string
+	region           string
+	ipAddress        string
+	dbUsername       string
+	dbUserPassword   string
+	connectionString string
+	store            store.AtlasClusterQuickStarter
+}
+
+func (opts *Opts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *Opts) Run() error {
+	if _, err := opts.store.CreateCluster(opts.newCluster()); err != nil {
+		return err
+	}
+
+	// Add IP to project’s IP access list
+	entry := opts.newWhitelist()
+	if _, err := opts.store.CreateProjectIPAccessList(entry); err != nil {
+		return err
+	}
+
+	// Create DBUser
+	if er := opts.createDatabaseUser(); er != nil {
+		return er
+	}
+
+	fmt.Println("Creating your cluster...")
+	if er := opts.Watch(opts.watcher); er != nil {
+		return er
+	}
+
+	// Get cluster's connection string
+	cluster, err := opts.store.AtlasCluster(opts.ConfigProjectID(), opts.clusterName)
+	if err != nil {
+		return err
+	}
+	opts.connectionString = cluster.SrvAddress
+
+	fmt.Printf(quickstartTemplate, opts.dbUsername, opts.dbUserPassword, opts.connectionString)
+	return nil
+}
+
+func (opts *Opts) watcher() (bool, error) {
+	result, err := opts.store.AtlasCluster(opts.ConfigProjectID(), opts.clusterName)
+	if err != nil {
+		return false, err
+	}
+	return result.StateName == "IDLE", nil
+}
+
+func (opts *Opts) createDatabaseUser() error {
+	user, err := opts.store.DatabaseUser(convert.AdminDB, opts.ConfigProjectID(), opts.dbUsername)
+	if err != nil {
+		if !strings.Contains(err.Error(), fmt.Sprintf("No user with username %s exists.", opts.dbUsername)) {
+			return err
+		}
+	}
+
+	if user != nil {
+		return nil
+	}
+
+	// Create dbUser
+	if _, err := opts.store.CreateDatabaseUser(opts.newDatabaseUser()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (opts *Opts) newDatabaseUser() *atlas.DatabaseUser {
+	return &atlas.DatabaseUser{
+		Roles:        convert.BuildAtlasRoles([]string{atlasAdmin}),
+		GroupID:      opts.ConfigProjectID(),
+		Password:     opts.dbUserPassword,
+		X509Type:     none,
+		AWSIAMType:   none,
+		LDAPAuthType: none,
+		DatabaseName: convert.AdminDB,
+		Username:     opts.dbUsername,
+	}
+}
+
+// newIPAddress returns client's public ip
+func newIPAddress() (string, error) {
+	publicIP := ""
+	for _, uri := range APIURIs {
```

Comment:
No, I didn’t. The only reason why I am iterating over this list is to make the function more reliable so that, if the first link of the list returns an error, this function would still work.